packages/inngest/src/components/Inngest.ts

```diff
@@ -169,6 +174,12 @@ export class Inngest<TClientOpts extends ClientOptions = ClientOptions>
 
   private _appVersion: string | undefined;
 
+  /**
+   * @internal
+   * Flag set by metadataMiddleware to enable step.metadata()
+   */
+  public _experimentalMetadataEnabled = false;
```

Comment:
Can we make this `protected` or `private` instead?

- `protected` would mean it's still typed as a `boolean` for the user
- `private` would resolve to `any` for the user

Either one ensures it's not exposed directly if we're initially using middleware to enable the feature.

Also, that means there's no need for the `_`; it can just be `experimentalMetadataEnabled`.